owncloud/core
# [WIP] Implement persistent major version workflow

## Description


TODO @mrow4a 

## Related Issue




- https://github.com/owncloud/enterprise/issues/5286

## Remaining To-Dos

- [x] Clean Up @JammingBen 
- [x] Improve layout for current @JammingBen 
- [x] Adjust restore to create new version @JammingBen 
- [x] Bind functionally completely to the `file_storage.save_version_author` flag @JammingBen 
- [ ] Rename `file_storage.save_version_author` flag @mrow4a 
- [x] Fix expiry error logs @JammingBen 
- [ ] Make sure the feature is backwards compatible with past current version logic @mrow4a 
- [ ] Add tests @mrow4a 
- [ ] ? @mrow4a 
- [ ] changelog @mrow4a 

## How tested

@mrow4a 

- prepare env
```
docker run -t -d -p 8098:9980 -e "extra_params=--o:ssl.enable=false" -e "username=admin" -e "password=admin" --name collaboraocdev --cap-add MKNOD collabora/code:6.4.8.6

occ config:app:set richdocuments wopi_url --value http://localhost:8098
occ a:d files_texteditor
occ a:e testing
occ a:e richdocuments

occ config:system:set file_storage.save_version_author --value true --type boolean

export OC_PASS=test
occ user:add --email test1@test.com --password-from-env test1
occ user:add --email test2@test.com --password-from-env test2

occ group:add grp1
occ group:add-member grp1 --member test1
occ group:add-member grp1 --member test2
```
- create shared folder with a document, and collaborativelly edit between test1 and test2
- check versions tab of the edited file in various editing configurations, restores and publishes

## Types of changes
